torch/_inductor/codegen/triton.py

```diff
@@ -97,10 +100,21 @@ def gen_common_triton_imports():
     return imports.getvalue()
 
 
+class ConstExprMin(sympy.Min):
+    """
+    Variant of sympy.Min that propagates triton constexprs.
+    While the sympy.Min maps to tl.minimum, this maps to an arithmetic macro.
+    https://github.com/triton-lang/triton/issues/3815
+    """
+
+    def macro(self, a: str, b: str) -> str:
+        return f"({a} * ({a} <= {b}) + {b} * ({b} < {a}))"
```

Comment:
Why not just put this directly in `_print_Min` (and `_print_Max`) -- is there a reason to keep the old behavior?

Reply:
Do you mean we should always print this macro, as opposed to `tl.minimum` and `tl.maximum`? I think it would work. I guess the only downside is that the code it generates is pretty ugly. Do you think it's worth it for the flexibility of supporting constexpr?

Reply:
If it generated faster code we should switch to it.